tests/src/python/test_qgsvectorlayer.py

```diff
@@ -2758,36 +2758,44 @@ def testFeatureRequestWithReprojectionAndVirtualFields(self):
 
     def testPrecision(self):
         layer = QgsVectorLayer("Polygon?crs=epsg:2056&field=pk:int", "vl", "memory")
-        layer.setGeometryPrecision(10)
+        geomOps = QgsVectorLayer.GeometryOptions()
+        geomOps.geometryPrecision = 10
+        layer.setGeometryOptions(geomOps)
```

Comment:
@nyalldawson, the effect of the new style is that we need 3 lines here instead of 1. I'm considering to revert for an easy to use API.

Reply:
Let's let @wonder-sk have the tie-breaker call?

Reply:
Then let's give him the full information, that I'll require change in precision in snap to grid on map tools.

```
connect ( layer, geometryOptionsChanged, this, updateSnapToGridPrecision );

void updateSnapToGridPrecision()
{
  GeometryOptions ops = layer->geometryOptions();
  if ( ops.geometryPrecision != mSnapToGridPrecision )
  {
    mSnapToGridPrecision = ops.geometryPrecision;
    reinitializeSnapToGrid();
  } 
}
```

vs.


```
connect ( layer, geometryPrecisionChanged, this, updateSnapToGridPrecision );

void updateSnapToGridPrecision()
{
  mSnapToGridPrecision = layer->geometryPrecision();
  reinitializeSnapToGrid();
}
```